src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -973,6 +973,7 @@ def filter_and_paginate(
             ValueError: if the filtered page number is out of bounds.
             RuntimeError: if the schema does not have a `to_model` method.
         """
+        query = query.distinct()
```

Comment:
just for me to understand, why was this necessary only now?

Reply:
Tbh I don't really know. I encountered duplicate results once I allowed filtering tags by prefix (e.g. `tag="startswith:t"`), and if one entity contained multiple tags that matched this filter it appeared more than once in the result.